controllers/pkg/reconcilers/spire-bootstrap/reconciler.go

```diff
@@ -0,0 +1,475 @@
+	crbAuthDelegator := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "spire-agent-tokenreview-binding",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      spirekubeconfig,
+				Namespace: "spire",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "system:auth-delegator",
+		},
+	}
+	if err := client.Create(ctx, crbAuthDelegator); err != nil {
+		return fmt.Errorf("failed to create ClusterRoleBinding (auth-delegator): %v", err)
+	}
+
+	// Create ClusterRoleBinding for pod-reader
+	crbPodReader := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "spire-agent-pod-reader-binding",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      spirekubeconfig,
+				Namespace: "spire",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "pod-reader",
+		},
+	}
+	if err := client.Create(ctx, crbPodReader); err != nil {
+		return fmt.Errorf("failed to create ClusterRoleBinding (pod-reader): %v", err)
+	}
+
+	// Create Secret
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "agent-sa-secret",
+			Namespace: "spire",
+			Annotations: map[string]string{
+				"kubernetes.io/service-account.name": spirekubeconfig,
+			},
+		},
+		Type: v1.SecretTypeServiceAccountToken,
+	}
+	if err := client.Create(ctx, secret); err != nil {
+		return fmt.Errorf("failed to create Secret: %v", err)
+	}
+
+	return nil
+}
+
+func (r *reconciler) createKubeconfigConfigMap(ctx context.Context, clientset *kubernetes.Clientset, clustername string) (*v1.ConfigMap, error) {
+
+	cmName := types.NamespacedName{Name: "kubeconfigs", Namespace: "spire"}
+	restrictedKC := &v1.ConfigMap{}
+	err := r.Get(ctx, cmName, restrictedKC)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get existing ConfigMap: %v", err)
+	}
+
+	// Retrieve the ServiceAccount token
+	secret, err := clientset.CoreV1().Secrets("spire").Get(ctx, "agent-sa-secret", metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ServiceAccount token: %v", err)
+	}
+	token := string(secret.Data["token"])
+
+	// Retrieve the cluster's CA certificate
+	configMap, err := clientset.CoreV1().ConfigMaps("kube-system").Get(ctx, "kube-root-ca.crt", metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster CA certificate: %v", err)
+	}
+	caCert := configMap.Data["ca.crt"]
+	caCertEncoded := strings.TrimSpace(base64.StdEncoding.EncodeToString([]byte(caCert)))
+
+	config := KubernetesConfig{
+		APIVersion: "v1",
+		Kind:       "Config",
+		Clusters: []Cluster{
+			{
+				Name: clustername,
+				Cluster: ClusterDetail{
+					CertificateAuthorityData: caCertEncoded,
+					Server:                   clientset.RESTClient().Get().URL().String(),
```

Comment:
clientset is needed to get the k8s API URL of the other remote clusters, so that we could do a workload attestation. I'm not sure how we would do the same if we used nephio capi client.